src/coreclr/vm/codeman.h

```diff
@@ -608,44 +610,926 @@ class UnwindInfoTable {
+                            {
+                                foundMeaningfulValue = true;
+                                break;
+                            }
+                        }
+
+                        if (foundMeaningfulValue)
+                            break;
+                        
+                        // This level is completely empty. Free it, and then null out the pointer to it.
+                        pointerToLevelData->Uninstall();
+                        free((void*)rawData);
+                    }
+                }
+
+                addressToPrepForCleanup = IncrementAddressByMaxSizeOfFragment(addressToPrepForCleanup);
+            }
+
+            // Free the array of fragments
+            delete pRangeSectionToCleanup;
+            free(pRangeSectionFragmentToFree);
+        }
+    }
+#endif // DACCESS_COMPILE
+
+#ifdef DACCESS_COMPILE
+    void EnumMemoryRangeSectionMapLevel(CLRDataEnumMemoryFlags flags, RangeSectionFragmentPointer& fragmentPointer, RangeSectionLockState* pLockState)
+    {
+        PTR_RangeSectionFragment fragment = fragmentPointer.VolatileLoadWithoutBarrier(pLockState);
+        while (fragment != NULL)
+        {
+            if (!DacEnumMemoryRegion(dac_cast<TADDR>(fragment), sizeof(RangeSectionFragment)))
+                return;
+
+            fragment->pRangeSection->EnumMemoryRegions(flags);
+
+            fragment = fragment->pRangeSectionFragmentNext.VolatileLoadWithoutBarrier(pLockState);
+        }
+    }
+
+    void EnumMemoryRangeSectionMapLevel(CLRDataEnumMemoryFlags flags, RangeSectionL1& level, RangeSectionLockState* pLockState)
+    {
+        if (!DacEnumMemoryRegion(dac_cast<TADDR>(&level), sizeof(level)))
+            return;
+
+        for (uintptr_t i = 0; i < entriesPerMapLevel; i++)
+        {
+            if (!level[i].IsNull())
+            {
+                EnumMemoryRangeSectionMapLevel(flags, level[i], pLockState);
+            }
+        }
+    }
+
+    template<class T>
+    void EnumMemoryRangeSectionMapLevel(CLRDataEnumMemoryFlags flags, T& level, RangeSectionLockState* pLockState)
+    {
+        if (!DacEnumMemoryRegion(dac_cast<TADDR>(&level), sizeof(level)))
+            return;
+
+        for (int i = 0; i < entriesPerMapLevel; i++)
+        {
+            if (level[i].IsNull())
+            {
+                EnumMemoryRangeSectionMapLevel(flags, *level[i].VolatileLoad(pLockState), pLockState);
+            }
+        }
+    }
+
+    void EnumMemoryRegions(CLRDataEnumMemoryFlags flags)
+    {
+        if (!DacEnumMemoryRegion(dac_cast<TADDR>(this), sizeof(*this)))
+            return;
+
+        // Always assume we are locked when enumerating
+        RangeSectionLockState lockState = RangeSectionLockState::ReaderLocked;
+        EnumMemoryRangeSectionMapLevel(flags, GetTopLevel(), &lockState);
+    }
+#endif// DACCESS_COMPILE
+
+};
+
+struct RangeSectionMapData
+{
+    BYTE Data[sizeof(RangeSectionMap)];
```

Comment:
```suggestion
    TADDR Data[sizeof(RangeSectionMap) / sizeof(TADDR)];
```
We need to make sure that the compiler places the static at sufficiently aligned address.